src/Data/Text/Lines/Internal.hs

```diff
@@ -243,6 +244,25 @@ lines (TextLines (Text arr off len) nls) = go off (U.toList nls)
 splitAtLine :: HasCallStack => Word -> TextLines -> (TextLines, TextLines)
 splitAtLine k = splitAtPosition (Position k 0)
 
+-- | Get line with given 0-based index, O(1).
+-- The resulting Text does not contain newlines.
+-- Returns "" if the line index is out of bounds.
```

Comment:
One design opportunity is to retain newlines in place, so that you can distinguish between an existing but empty line `"\n"` and line out of bounds `""`.

Reply:
On Windows with `\r\n` line endings it feels undesirable to scrap `\n` but not `\r`, so maybe it's better to keep `\n` in place.

Reply:
Hmm, right, I didn't think about windows.
Don't you think the quickcheck properties make sense?
I implemented getLine in a way that is consistent with the results that `lines` function gives.
So which also doesn't strip the \r:
```
-- >>> lines "foo\r\nbar"
-- ["foo\r","bar"]
```

Reply:
Fair enough, consistency with `lines` makes sense.